apache2/apache2_util.c

```diff
@@ -284,12 +284,27 @@ static void internal_log_ex(request_rec *r, directory_config *dcfg, modsec_rec *
         }
         else hostname = "";
 
-#if AP_SERVER_MAJORVERSION_NUMBER > 1 && AP_SERVER_MINORVERSION_NUMBER > 2
+/* Non-standalone modules use amended format string for logging */
+#if !(defined(VERSION_STANDALONE))
+  #if AP_SERVER_MAJORVERSION_NUMBER > 1 && AP_SERVER_MINORVERSION_NUMBER > 2
 	ap_log_rerror(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r,
             "ModSecurity: %s%s [uri \"%s\"]%s", str1, hostname, log_escape(msr->mp, r->uri), unique_id);
-#else
+  #else
         ap_log_error(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r->server,
             "ModSecurity: %s%s [uri \"%s\"]%s", str1, hostname, log_escape(msr->mp, r->uri), unique_id);
+  #endif
+/* Standalone module must use original format string for logging with explicit
+ * "[client %s]" to log client IP address (no Apache to implicitly add this) */
+#else
+  #if AP_SERVER_MAJORVERSION_NUMBER > 1 && AP_SERVER_MINORVERSION_NUMBER > 2
+	ap_log_rerror(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r,
+            "[client %s] ModSecurity: %s%s [uri \"%s\"]%s", r->useragent_ip ? r->useragent_ip : r->connection->client_ip, str1,
+            hostname, log_escape(msr->mp, r->uri), unique_id);
+  #else
+        ap_log_error(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r->server,
+                "[client %s] ModSecurity: %s%s [uri \"%s\"]%s", msr->remote_addr ? msr->remote_addr : r->connection->remote_ip, str1,
```

Comment:
Just for the better visibility, could you use here the same indent (and below in line [306](https://github.com/owasp-modsecurity/ModSecurity/pull/3374/files#diff-f7042e3212265ddbdf32d44c0fca670fbe6d8be64f688aa7dead0c25151c0669R306) as in line [301/302](https://github.com/owasp-modsecurity/ModSecurity/pull/3374/files#diff-f7042e3212265ddbdf32d44c0fca670fbe6d8be64f688aa7dead0c25151c0669R301)?

Reply:
Thanks, well spotted. Indentation is now fixed.